### PR TITLE
Internalization of OneToOne and ManyToOne Column classes

### LINKED ISSUE #2631: Internalization of OneToOne and ManyToOne Column classes
https://github.com/dotnet/machinelearning/blob/25404b83df1b6bf9f548ee59396a87e43f0f9058/src/Microsoft.ML.Data/Transforms/ColumnBindingsBase.cs#L17

https://github.com/dotnet/machinelearning/blob/25404b83df1b6bf9f548ee59396a87e43f0f9058/src/Microsoft.ML.Data/Transforms/ColumnBindingsBase.cs#L106

https://github.com/dotnet/machinelearning/blob/25404b83df1b6bf9f548ee59396a87e43f0f9058/src/Microsoft.ML.Data/Transforms/ColumnBindingsBase.cs#L112

Should be hidden

### PR DESCRIPTION
**Pull request by codemzs**
fixes #2631

### CODE REVIEW

File: src/Microsoft.ML.Transforms/GcnTransform.cs [hunk 1]
```diff
@@ -77,6 +77,7 @@ internal sealed class GcnOptions : TransformInputBase
             public float Scale = LpNormalizingEstimatorBase.Defaults.Scale;
         }
 
+        [BestFriend]
```
Comment (yaeldekel) [inline src/Microsoft.ML.Transforms/GcnTransform.cs]: >BestFriend [](start = 9, length = 10)

Why is this needed?

Comment (codemzs) [reply src/Microsoft.ML.Transforms/GcnTransform.cs]: You are right, its not needed here.
